mqtt-impl/src/main/java/io/streamnative/pulsar/handlers/mqtt/TopicSubscriptionManager.java

```diff
@@ -49,7 +49,7 @@ public void removeAllConsumer() {
     }
 
     public CompletableFuture<Void> unsubscribe(Topic topic, boolean cleanSubscription) {
-        Pair<Subscription, Consumer> subscriptionConsumerPair = topicSubscriptions.get(topic);
+        Pair<Subscription, Consumer> subscriptionConsumerPair = topicSubscriptions.remove(topic);
```

Comment:
This may cause some issue, when pulsar unsubscribe fail.

Reply:
yes, rollback